File: python/maplestory_openapi/api/dto/notice/update_notice_list.py
from datetime import datetime
from pydantic import BaseModel


class UpdateNoticeListItem(BaseModel):
    """
    업데이트 공지

    title(str): 공지 제목
    url(str): 공지 링크
    notice_id(int): 공지 식별자
    date(datetime): 공지 등록일
    """
    title: str
    url: str
    notice_id: int
    date: datetime


class UpdateNoticeList(BaseModel):
    """
    업데이트 목록

    update_notice(list[UpdateNoticeListItem]): 공지 목록
    """
    update_notice: list[UpdateNoticeListItem]

File: js/src/maplestory/api/dto/notice/updateNoticeList.ts
import { UpdateNoticeListBody, UpdateNoticeListItemBody } from '../../response/notice/updateNoticeListBody';

/**
 * 업데이트 목록
 */
export class UpdateNoticeListDto {

  /**
   * 공지 목록
   */
  updateNotice: UpdateNoticeListItemDto[];

  constructor(obj: UpdateNoticeListBody) {
    const { update_notice } = obj;

    this.updateNotice = update_notice.map(notice => new UpdateNoticeListItemDto(notice));
  }
}

/**
 * 업데이트 공지
 */
export class UpdateNoticeListItemDto {

  /**
   * 공지 제목
   */
  title: string;

  /**
   * 공지 링크
   */
  url: string;

  /**
   * 공지 식별자
   */
  noticeId: number;

  /**
   * 공지 등록일
   */
  date: Date;

  constructor(obj: UpdateNoticeListItemBody) {
    const { title, url, notice_id, date } = obj;

    this.title = title;
    this.url = url;
    this.noticeId = notice_id;
    this.date = new Date(date);
  }
}
